db/api/tests/endpoints/projects.test.js

```diff
@@ -0,0 +1,373 @@
+                        expect.objectContaining(EXPECTED_PROJECT2),
+                        expect.objectContaining(EXPECTED_PROJECT3)
+                    ])
+                );
+            });
+    });
+
+    it('able to get all Projects', async () => {
+        await request(app)
+            .get(ENDPOINT)
+            .expect(200)
+            .then((res) => {
+                expect(res.body).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining(EXPECTED_PROJECT1),
+                        expect.objectContaining(EXPECTED_PROJECT2),
+                        expect.objectContaining(EXPECTED_PROJECT3)
+                    ])
+                );
+                expect(res.body).toHaveLength(3);
+            });
+    });
+
+    it('able to filter by Sponsor ID', async () => {
+        await request(app)
+            .get(ENDPOINT + '?sponsor_id=2')
+            .expect(200)
+            .then((res) => {
+                expect(res.body).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining(EXPECTED_PROJECT3)
+                    ])
+                );
+                expect(res.body).toHaveLength(1);
+            });
+    });
+
+    it('returns bad request (HTTP 400) when finding with an invalid argument', async () => {
+        await request(app)
+            .get(ENDPOINT + '?sponsor_id=a')
+            .expect(400)
+            .then((res) => {
+                expect(res.body['errors']).toHaveLength(1);
+                expect(res.body['errors'][0]).toHaveProperty('problem', 'Failed to validate the argument "a" for the parameter "sponsor_id"');
+                expect(res.body['errors'][0]).toHaveProperty('reason', 'Expected a number between 1 and 2147483647');
+            });
+    });
+
+    it('able to filter by Project Name', async () => {
+        await request(app)
+            .get(ENDPOINT + '?name=Lemur%20Protection')
+            .expect(200)
+            .then((res) => {
+                expect(res.body).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining(EXPECTED_PROJECT2)
+                    ])
+                );
+                expect(res.body).toHaveLength(1);
+            });
+    });
+
+    it('filters by Project Name is case insensitive', async () => {
+        await request(app)
+            .get(ENDPOINT + '?name=lEmUR%20PRoTECtIon')
+            .expect(200)
+            .then((res) => {
+                expect(res.body).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining(EXPECTED_PROJECT2)
+                    ])
+                );
+                expect(res.body).toHaveLength(1);
+            });
+    });
+
+    it('able to filter by Sponsor ID and Project Name', async () => {
+        await request(app)
+            .get(ENDPOINT + '?sponsor_id=1&name=Madagascar%20Reforesting%20Project')
```

Comment:
It is probably cleaner to use a url builder for many of these tests, I would suggest looking into this library:
https://nodejs.org/api/querystring.html

Reply:
Cool will do